### PR TITLE
Add edge cases to summon item spell, and tweak to feedback if object is hidden

### LINKED ISSUE #22400: Summon item works weird with containers
**Issue Description**:
Extended bugs and information from #10417 

When the item is:
- In a chest cavity and recalled, it recalls the item (This works as it should) But if you want to put something else in the cavity, you first have to empty it from the first item, which also teleports it back to the person.
- In a storage implant, it drops the item and the storage implant is deleted. It probably drops it alongside the item, but as it doesn't really count as a container, it just deletes itself probably

**What did you expect to happen**: 
To have summon item work consistently and not cause any weird teleportation issues with cavity implants.

**What happened instead**:
Deleted storage implants and strange cavity implants full of wormholes

**Steps to reproduce the problem**:
Bind an item to yourself with summon item, put that inside a skrell and then summon the item, which works fine. Then try to empty the chest cavity and see the item teleported twice

Put a bound item in a storage implant, see the storage delete itself upon recall, and drop the item.

**When did the problem start happening**:
Probably for a looong time.

### PR DESCRIPTION
**Pull request by Vi3trice**
## What Does This PR
The summon item spell still had some odd interactions and feedback, in order:
Cavities would not get cleared properly, it now does so.
Headpockets would just cause the spell to runtime due to being in nullspace, and gets its own check now.
Storage implants just got deleted by the spell, and now it just pulls the item out, leaving the implant in place.

Feedback for the disappearing item now has a check to see if it's intended to be seen. It doesn't make sense for people who can't even see the item to be told that X suddenly disappeared. The storage implant in particular gives no feedback unless its content are being viewed while the item is recalled.

## Why It's Good For The Game
Fixes #22400. More sensible feedback based on what happens is better as well.

## Images of changes
Summon-Cavity
Summon-Pocket

## Testing
Used summon item on an object that was stored in a headpocket, cavity, and storage implant.

## Changelog
:cl:
fix: Instant Summons can recall items out of headpockets
fix: Instant Summons properly recalls items in body cavities
fix: Instant Summons does not delete storage implants anymore
/:cl:

### CODE REVIEW

File: code/datums/spells/summonitem.dm [hunk 1]
```diff
@@ -55,9 +55,22 @@
...
 			var/infinite_recursion = 0 //I don't want to know how someone could put something inside itself but these are wizards so let's be safe
 
 			while(!isturf(item_to_retrieve.loc) && infinite_recursion < 10) //if it's in something you get the whole thing.
+				if(istype(item_to_retrieve.loc, /obj/item/organ/internal/headpocket))
+					var/obj/item/organ/internal/headpocket/pocket = item_to_retrieve.loc
+					if(pocket.owner)
+						to_chat(pocket.owner, "<span class='warning'>Your [pocket.name] suddenly feels lighter. How strange!</span>")
+					visible_item = FALSE
+					break
+				if(istype(item_to_retrieve.loc,/obj/item/storage/hidden/implant)) //The implant should be left alone
```
Comment (DGamerL) [inline code/datums/spells/summonitem.dm:68]:
```suggestion
				if(istype(item_to_retrieve.loc, /obj/item/storage/hidden/implant)) //The implant should be left alone
```

File: code/datums/spells/summonitem.dm [hunk 2]
```diff
@@ -55,9 +55,22 @@
...
+					var/obj/item/storage/S = item_to_retrieve.loc
+					for(var/mob/M in S.mobs_viewing)
+						to_chat(M, "<span class='warning'>[item_to_retrieve] suddenly disappears!</span>")
```
Comment (DGamerL) [inline code/datums/spells/summonitem.dm:71]: Will this only affect the storage implant owner? Not sure how `mobs_viewing` works

**Review by DGamerL — COMMENTED** [review]
Very neat PR, thanks for fixing this!

File: code/datums/spells/summonitem.dm [hunk 3]
```diff
@@ -78,6 +91,11 @@
 								if(!C.has_embedded_objects())
 									C.clear_alert("embeddedobject")
 								break
+							if(item_to_retrieve == part.hidden)
+								visible_item = FALSE
+								part.hidden = null
+								to_chat(C, "<span class='warning'>Your [part.name] suddenly feels emptier. How weird!</span>")
```
[the remark this answers is not in the document]
Comment (Vi3trice) [reply code/datums/spells/summonitem.dm:97]: That's what `part` would have done over `part.name`.

File: code/datums/spells/summonitem.dm (hunk 2 above)
Comment (Vi3trice) [reply code/datums/spells/summonitem.dm:71]: Generally speaking, it'd be anyone looking into the storage. If the owner isn't looking into the implant, then they won't get a clue of what happened.

File: code/datums/spells/summonitem.dm (hunk 3 above)
[the remark this answers is not in the document]
Comment (DGamerL) [reply code/datums/spells/summonitem.dm:97]: Thanks, didn't know that!

**Comment by 1080pCat** [comment]
> Isntant

File: code/datums/spells/summonitem.dm [hunk 4]
```diff
@@ -55,9 +55,22 @@
...
 			var/infinite_recursion = 0 //I don't want to know how someone could put something inside itself but these are wizards so let's be safe
 
 			while(!isturf(item_to_retrieve.loc) && infinite_recursion < 10) //if it's in something you get the whole thing.
+				if(istype(item_to_retrieve.loc,/obj/item/organ/internal/headpocket))
+					var/obj/item/organ/internal/headpocket/pocket = item_to_retrieve.loc
+					if(pocket.owner)
+						to_chat(pocket.owner, "<span class='warning'>Your [pocket.name] suddenly feels lighter. How strange!</span>")
+					visible_item = FALSE
+					break
+				if(istype(item_to_retrieve.loc,/obj/item/storage/hidden/implant)) //The implant should be left alone
```
Comment (DGamerL) [inline code/datums/spells/summonitem.dm:68]:
```suggestion
				if(istype(item_to_retrieve.loc, /obj/item/storage/hidden/implant)) //The implant should be left alone
```

File: code/datums/spells/summonitem.dm [hunk 5]
```diff
@@ -55,9 +55,22 @@
 
 		else	//Getting previously marked item
 			var/obj/item_to_retrieve = marked_item
+			var/visible_item = TRUE //Items that silently disappear will have the message suppressed
 			var/infinite_recursion = 0 //I don't want to know how someone could put something inside itself but these are wizards so let's be safe
 
 			while(!isturf(item_to_retrieve.loc) && infinite_recursion < 10) //if it's in something you get the whole thing.
+				if(istype(item_to_retrieve.loc,/obj/item/organ/internal/headpocket))
```
Comment (DGamerL) [inline code/datums/spells/summonitem.dm:62]:
```suggestion
				if(istype(item_to_retrieve.loc, /obj/item/organ/internal/headpocket))
```

**Review by DGamerL — COMMENTED** [review]
It seems like something went wrong with committing my review?

**Review by S34NW — APPROVED** [review]

**Pull request merged**